mindsdb/libs/phases/stats_generator/stats_generator.py

```diff
@@ -223,32 +181,202 @@ def getParamsAsDictionary(self, params):
+            ,'cummulative_z_score': len(z_score_outlier_indexes)/len(columns[col_name])
+        }
+        return data
+
+    def lof_score(self, stats, columns, col_name):
+        if stats[col_name][KEYS.DATA_TYPE] != DATA_TYPES.NUMERIC:
+            return {}
+
+        np_col_data = np.array(columns[col_name]).reshape(-1, 1)
+        lof = LocalOutlierFactor(contamination='auto')
+        outlier_scores = lof.fit_predict(np_col_data)
+        outlier_indexes = [i for i in range(len(columns[col_name])) if outlier_scores[i] < -0.8]
+
+        return {
+            'lof_outliers': outlier_indexes
+            ,'cummulative_lof_score': len(outlier_indexes)/len(columns[col_name])
+        }
+
+
+    def similariy_score(self, stats, columns, col_name):
+        col_data = columns[col_name]
+
+        similarities = []
+        for _, other_col_name in enumerate(columns):
+            if other_col_name == col_name:
+                continue
+            else:
+                similarity = matthews_corrcoef(list(map(str,col_data)), list(map(str,columns[other_col_name])))
+                similarities.append((other_col_name,similarity))
+
+        return {
+            'similarities': similarities
+            ,'similarity_score': max(map(lambda x: x[1], similarities))
+        }
 
 
+    def clf_based_correlation_score(self, stats, columns, col_name):
+        full_col_data = columns[col_name]
+
+        dt_clf = DecisionTreeClassifier()
+
+        other_feature_names = []
+        other_features = []
+        for _, other_col_name in enumerate(columns):
+            if other_col_name == col_name:
+                continue
+
+            other_feature_names.append(other_col_name)
+            le = LabelEncoder()
+            _stringified_col = list(map(str,columns[other_col_name]))
+            le.fit(_stringified_col)
+            other_features.append(list(le.transform(_stringified_col)))
+
+        other_features_t = np.array(other_features, dtype=object).transpose()
+
+        le = LabelEncoder()
+        _stringified_col = list(map(str,full_col_data))
+        le.fit(_stringified_col)
+        y = le.transform(_stringified_col)
+        dt_clf.fit(other_features_t,y)
+        prediction_score = dt_clf.score(other_features_t,y)
+        corr_scores = list(dt_clf.feature_importances_)
+        highest_correlated_column = max(corr_scores)
+        return {
+            'correlation_score': prediction_score * highest_correlated_column
+            ,'highest_correlation': max(corr_scores)
+            ,'most_correlated_column': other_feature_names[corr_scores.index(max(corr_scores))]
+        }
+
+    def data_quality_score(self, stats, columns, col_name):
+        scores_used = 0
+        scores_total = 0
+        scores = ['correlation_score', 'cummulative_lof_score', 'cummulative_z_score', 'data_type_distribution_score', 'empty_cells_score', 'duplicate_score', 'similarity_score', 'value_distribution_score']
```

Comment:
question: are all these scores uniformly normalized/scaled (example: all possible values of all scores are floats between 0 to 1)? are all these scores such that the scales are mapped to 1 means something like very bad, 0 means something like very good (example is correlation_score = 1 a bad thing just as similarity_score=1 also a bad thing)?

Reply:
All scores should be normalized between 0 and 1, where 1 is worst and 0 is best, I'll make sure once again that it applies to all of them.

Reply:
Confirmed that all the scores follow the pattern where 0 is good and 1 is bad, and added comments to all the scoring functions in #0cd13921838fef53140677f9fd47618154508703